astropy/io/misc/parquet.py

```diff
@@ -513,3 +513,149 @@ def get_pyarrow():
         writer_version = "2.0"
 
     return pa, parquet, writer_version
+
+
+def write_parquet_votable(table, output, *, metadata, overwrite=False):
+    """
+    Writes a Parquet file with a VOT (XML) metadata table included.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        Data table that is to be written to output.
+    output : str or path-like
+        The filename to write the table to.
+    metadata : dict
+        Nested dictionary (keys = column names; sub-keys = meta keys) for each
+        of the columns containing a dictionary with metadata.
+
+    """
+    import io
+    import xml.etree.ElementTree
+
+    import pyarrow.parquet
+
+    from astropy.io.votable.tree import VOTableFile
+
+    if os.path.exists(output):
+        if overwrite:
+            # We must remove the file prior to writing below.
+            os.remove(output)
+        else:
+            raise OSError(NOT_OVERWRITING_MSG.format(output))
+
+    # Prepare the VOTable (XML)
+    # We only use the first row of the astropy table to get the general
+    # information such as arraysize, ID, or datatype.
+    votablefile = VOTableFile()
+    votable_write = votablefile.from_table(table[0:1])
+
+    # Then add the other metadata keys to the FIELDS parameters of the VOTable
+    metadatakeys = list(metadata[next(iter(metadata.keys()))].keys())
+    for field in votable_write.resources[0].tables[0].fields:
+        for mkey in metadatakeys:
+            if mkey in field._attr_list:
+                setattr(field, mkey, metadata[field.name][mkey])
+            else:
+                if (mkey == "description") & (field.description is not None):
+                    field.description = metadata[field.name]["description"]
+                else:
+                    print(f"Warning: '{mkey}' is not a valid VOT metadata key")
+
+    # Convert the VOTable object into a Byte string to create an
+    # XML that we can add to the Parquet metadata
+    xml_bstr = io.BytesIO()
```

Comment:
Does this need to be a byte string?  The votable content will already be utf-8.  Are the bytes needed for parquet?